libs/island-ui/core/src/lib/Checkbox/Checkbox.tsx

```diff
@@ -52,6 +52,7 @@ export const Checkbox = ({
   dataTestId,
   filled = false,
 }: CheckboxProps & TestSupport) => {
+  const [isChecked, setIsChecked] = useState(checked || false)
```

Comment:
I see you're intentionally trying to use this component as uncontrolled.

We should try to emulate React's design for controlled and uncontrolled inputs (checked vs defaultChecked props). Maybe this guide will help:

https://maxschmitt.me/posts/react-components-controlled-uncontrolled/

Reply:
That pattern looks good. I added also defaultChecked since we are dealing with checkbox which can either have a string value or boolean value for checked. We need the checkbox value to either be a string or bool string in the query string.

Commit here tell me what you think
https://github.com/island-is/island.is/pull/10123/commits/017323bf997412189a4cc3227abc04b27a300b4e

Reply:
Better!

But I don't understand why we need defaultValue or any fancy logic around "value". Afaik checkboxes should only use `checked` (or `defaultChecked`). There is only one `value` which is used when the form is serialized and/or submitted. But maybe our checkboxes are different?

Let's look at this on Monday.